docs/Examples/Middleware/Main.purs

```diff
@@ -32,8 +32,8 @@ headerMiddleware :: (HTTPure.Request -> HTTPure.ResponseM) ->
                     HTTPure.Request ->
                     HTTPure.ResponseM
 headerMiddleware router request = do
-  response <- router request
-  HTTPure.response' response.status (header <> response.headers) response.body
+  response@{ headers } <- router request
+  pure $ response { headers = header <> headers }
```

Comment:
Wow this is black magic, I did not know you could do this in purescript :+1: :+1: 

... I actually like this a whole lot more than using the `response'` helpers, at least for this example...